### ISSUE TITLE
Hello world samples ?

### ISSUE THREAD

**Issue by maxired** [nginx/unit#8]
Hey,

this project looks very promising, but I got hard time trying to understand what you expect an application to look like.

Is there a place where I could find hello world application for the different supported language?

**Comment by defanator** [comment]
@maxired you can find "hello world" examples for PHP, Python and Go in pre-compiled packages:
http://unit.nginx.org/docs-installation.html#precompiled-packages

(Also, we're going to add package sources to this repo as well.)

**Comment by maxired** [comment]
@defanator thanks.
so from what I could read in examples files of http://nginx.org/packages/mainline/ubuntu/pool/nginx/u/unit/unit_0.1-1~xenial_amd64.deb

For php : just write regular php that would be executed on any webserver

For pyhton: use wsgi for now. Here Helo world from wikipedia
```python
def application(environ, start_response):
    start_response('200 OK', [('Content-Type', 'text/plain')])
    yield 'Hello World\n'
```

For go: this is different story. you should write  an app, creating a `net/http` server, and bind it with `unit.ListenAndServe("8000", nil)` . Does this port value mean anything ?

**Comment by defanator** [comment]
@maxired no, in case of plugging Go app into unit you may use any address/port there. In order to build and test "hello world" Go app from examples provided in the package, please do the following:

```
# sudo apt-get install unit-dev
# go build -o /tmp/go-app /usr/share/doc/unit/examples/go-app/let-my-people.go
# sudo service unitd restoreconfig /usr/share/doc/unit/examples/example.config
# curl http://localhost:8500/foo/bar/
Method : GET
URL    : /foo/bar/
Host   : localhost:8500
```

**Comment by docteurklein** [comment]
yould you have an example application for php as well?

**Comment by maxired** [comment]
@docteurklein, looks like you can't just write any php code as you would with apache or any other php cgi.

for example the sample in the repo is 
```php
<?php phpinfo(); ?>
```

**Comment by igorsysoev** [comment]
On 12 Sep 2017, at 12:48, Maxence Dalmais <notifications@github.com> wrote:
> @docteurklein, looks like you can't just write any php code as you would with apache or any other php cgi.
> 
No. You should be able to run any existent PHP code without changes.


-- 
Igor Sysoev
http://nginx.com

**Comment by docteurklein** [comment]
I can confirm it starts working on my legacy app.
here is a working setup for symfony:

```
{
    "listeners": {
        "*:8300": {
            "application": "api"
        }
    },
    "applications": {
        "api": {
            "type": "php",
            "workers": 20,
            "root": "/usr/src/app/web",
            "index": "app.php",
            "script": "app.php"
        }
    }
}
```

**Comment by centminmod** [comment]
Would love to see a official list of usage examples of Nginx Unit submitted by users especially for python and go apps which I am not that familiar with.

**Comment by rpv-tomsk** [comment]
Example is also needed for recently-added Perl support.

Thanks!

**Comment by defanator** [comment]
@rpv-tomsk here they are:

https://github.com/nginx/unit/blob/master/pkg/deb/debian.module/unit.example-perl-config
https://github.com/nginx/unit/blob/master/pkg/deb/debian.module/unit.example-perl-app

**Comment by VBart** [comment]
@rpv-tomsk there's one in the blog post: https://www.nginx.com/blog/unit-0-6-beta-release-available-now/

**Comment by lth2015** [comment]
@defanator @maxired 
when I use these command, it doesn't work:
```bash
# go build -o /tmp/go-app /usr/share/doc/unit/examples/go-app/let-my-people.go
# sudo service unitd restoreconfig /usr/share/doc/unit/examples/example.config
```
```json
{
	"success": "Reconfiguration done."
}
```
```bash
curl http://localhost:8500/foo/bar/
Read header timeout
```

Golang example doesn't work in my centos7 system, can you help me?

**Comment by lth2015** [comment]
Unit binary version and Unit source code version is not same, details in here: [https://github.com/nginx/unit/issues/126](https://github.com/nginx/unit/issues/126), close this issue.

**Comment by dac88** [comment]
@defanator  - would it be possible to get a POST example, ideally in Perl? thanks.

**Comment by sysdef** [comment]
@dac88

should work like that:

app.psgi

`package API;

use 5.24.0;
use strict;
use warnings;

my $app = sub {

  my $env = shift;

  warn( Dumper($env) );

  my $method = $env->{'REQUEST_METHOD'};
  my $uri    = $env->{'REQUEST_URI'};

  my $input;
  if ( $method eq 'POST' ) {
    $env->{'psgi.input'}->read( $input, 5000 );
  }

  # /ping
  if ( $method$uri eq 'POST/ping' ) {
    use Data::Dumper;
    return [
      '200',
      [ 'Content-Type' => 'text/plain' ],
      [ "POST: \n" . $input . "\nENV: " . Dumper($env) ],
    ];
  }

};
`

**Comment by dac88** [comment]
@sysdef  - much appreciated.

**Comment by gkapad** [comment]
Asimple php/html page :

`<!DOCTYPE html>`
`<link rel="stylesheet" type="text/css" href="/index.css">`
`<?php echo "Hello,  "; ?>`
`<span class="red">app1</span>`
`<pre><?php print_r($_SERVER); ?>`
`</pre>`

this kind of php code dont work. cuse 'index.css' is parse as 
Content-type:text/html
so the the browser dont apply it.
I do something wrong, or this is normal fot unit ???

----- index.css:
.red{ color:red;}

**Comment by VBart** [comment]
@gkapad It's not related to php code. Similar to php-fpm, Unit doesn't serve static files. Currently, you need to put static files to CDN or use nginx to serve them.

**Comment by dbuelow** [comment]
Static file support is available since v1.11.0. See [documentation](https://unit.nginx.org/configuration/) for configuration and a lot of examples.

@VBart: could be closed, right?

**Comment by VBart** [comment]
The "Hello world" samples in all supported languages are available here: https://unit.nginx.org/howto/samples/
